csvcubed/csvcubed/models/inspectsparqlresults.py

```diff
@@ -0,0 +1,209 @@
+"""
+Inspect SPARQL query results
+----------------------------
+"""
+
+from os import linesep
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Dict, List
+from csvcubed.utils.qb.components import get_printable_component_property
+from rdflib import URIRef
+
+from rdflib.query import ResultRow
+
+from csvcubed.utils.sparql import none_or_map
+from csvcubed.utils.printable import (
+    get_printable_list_str,
+    get_printable_tabular_list_str,
+    get_printable_tabular_str,
+)
+
+
+@dataclass()
+class CatalogMetadataSparqlResult:
+    """
+    Model to represent result of the `select_csvw_catalog_metadata` sparql query.
+    """
+
+    sparql_result: ResultRow
```

Comment:
I was more thinking that we'd do the mapping outside of this model so that this model could be used for more than just printing strings in the future.

Reply:
Perhaps we could bring the attribute definitions outside of `__post_init__` and then have a static method which takes a `ResultRow` and creates the `CatalogMetadataSparqlResult`? That way we're not exposing the `ResultRow` to anywhere that doesn't need to know about it - people can just reference the individual properties then.